lib/job/arUpdateEsActorRelationsJob.class.php

```diff
@@ -0,0 +1,116 @@
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Updates actor document relationships in the Elasticsearch index
+ *
+ * @package    symfony
+ * @subpackage jobs
+ */
+
+class arUpdateEsActorRelationsJob extends arBaseJob
+{
+  /**
+   * @see arBaseJob::$requiredParameters
+   */
+  protected $extraRequiredParameters = ['actorIds'];
+
+  public function runJob($parameters)
+  {
+    if (empty($parameters['actorIds']))
+    {
+      $this->error($this->i18n->__('Called arUpdateEsActorRelationsJob without specifying what needs to be updated.'));
+
+      return false;
+    }
+
+    $message = $this->i18n->__('Updating the relationships of %1 actor(s).', ['%1' => count($parameters['actorIds'])]);
+
+    $this->job->addNoteText($message);
+    $this->info($message);
+
+    $count = 0;
+    foreach ($parameters['actorIds'] as $id)
+    {
+      if (null === $object = QubitActor::getById($id))
+      {
+        $this->info($this->i18n->__('Invalid actor id: %1', ['%1' => $id]));
+
+        continue;
+      }
+      
+      // Don't count invalid description ids
+      $count++;
+
+      $this->updateActorRelationships($object);
+      $message = $this->i18n->__('Updated %1 actors(s).', ['%1' => $count]);
+
+      // Minimize memory use in case we're dealing with a large number of information objects
+      Qubit::clearClassCaches();
+
+      // Status update every 100 descriptions
+      if (0 == $count % 100)
+      {
+        $this->info($message);
+      }
+    }
+
+    // Final status update, if total count is not a multiple of 100
+    if (0 != $count % 100)
+    {
+      $this->info($message);
+    }
+
+    $this->job->setStatusCompleted();
+    $this->job->save();
+
+    return true;
+  }
+
+  private function updateActorRelationships($actor)
+  {
+    $relationData = arElasticSearchActorPdo::serializeObjectRelations($actor->id);
+    QubitSearch::getInstance()->partialUpdate($actor, ['actorRelations' => $relationData]);
+  }
+
+  public static function relationActorIds($actorId, $inclusive = true)
+  {
+    // Get actor's current relations
+    $relationData = arElasticSearchActorPdo::serializeObjectRelations($actorId);
+
+    // Parse parse out all actor IDs
+    $actors = [];
+
+    foreach($relationData as $relation)
+    {
+      $actors[] = $relation['objectId'];
+      $actors[] = $relation['subjectId'];
+    }
+
+    $actors = array_unique($actors);
+
+    // Omit ID of actor ID itself if not in inclusive mode (for deletes)
+    if (!$inclusive)
+    {
+      $actors = array_unique($actors, [$actorId]);
```

Comment:
Should this be `array_diff`? Also, it's really needed? The actor should be saved/deleted in all cases outside the job.

Reply:
I've simplified the logic and gotten rid of the inclusive flag. When deleting one needs to remove the ID of the actor being deleted from the array of related IDs, but I've just done this outside the function.